internal/app/backend/backend_service.go

```diff
@@ -117,22 +117,40 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 	return nil
 }
 
-func synchronizeSend(ctx context.Context, syncStream synchronizerStream, m *sync.Map, proposals <-chan *pb.Match) error {
-sendProposals:
+func synchronizeSend(ctx context.Context, syncStream synchronizerStream, m *sync.Map, pc <-chan *pb.Match) error {
```

Comment:
Doing this adds buffering until the mmf is finished, which isn't desired.

You could just use loadandstore, returning an error if the value already exists (everything will clean up once an error is return which cancels the context.

Reply:
But this won't return all of the duplicated IDs though.

Reply:
That doesn't matter.

Duplicate ids indicate a programming error.  We just need to fail loudly (because the evaluator won't work anyways) so that they know they're using OM wrong.  Having 1 duplicate id or a million doesn't change that they need to fix their programming.